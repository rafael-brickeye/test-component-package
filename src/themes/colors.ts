// All values used here come from Figma file design styling

const background = {
  white: '#ffffff',
};

const surface = {
  grey: '#C5CCD6',
};

const text = {
  black: '#000000DE',
  grey: '#71727A',
  white: '#ffffff',
};

const brand = {
  protect: {
    primary: {
      main: '#0048BB',
    },
  },
  lumicon: {
    primary: {
      light: '#FF9F34',
      main: '#FF8C4D',
      dark: '#FF5968',
    },
    secondary: {
      light: '#F46167',
      main: '#F05E40',
      dark: '#9E9E9E',
    },
    info: {
      light: '#64B6F7',
      main: '#2196F3',
      dark: '#0B79D0',
    },
    error: {
      light: '#F88078',
      main: '#F44336',
      dark: '#E31B0C',
    },
    warning: {
      light: '#FFB547',
      main: '#FF9800',
      dark: '#C77700',
    },
    success: {
      light: '#7BC67E',
      main: '#4CAF50',
      dark: '#3B873E',
    },
  },
};

const shadow = {
  default: {
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 1,
    },
    shadowOpacity: 0.18,
    shadowRadius: 1.0,
    elevation: 1,
  },
  wide: {
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.23,
    shadowRadius: 2.62,
    elevation: 4,
  },
};

export const color = {text, brand, surface, background, shadow};
