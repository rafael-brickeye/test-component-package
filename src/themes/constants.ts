import {TextStyle, ViewStyle} from 'react-native';

export const defaultViewStyles: Record<keyof Omit<ViewStyle, 'testID'>, string> = {
  alignContent: 'alignContent',
  alignItems: 'alignItems',
  alignSelf: 'alignSelf',
  aspectRatio: 'aspectRatio',
  backfaceVisibility: 'backfaceVisibility',
  backgroundColor: 'backgroundColor',
  borderBottomColor: 'borderBottomColor',
  borderBottomEndRadius: 'borderBottomEndRadius',
  borderBottomLeftRadius: 'borderBottomLeftRadius',
  borderBottomRightRadius: 'borderBottomRightRadius',
  borderBottomStartRadius: 'borderBottomStartRadius',
  borderBottomWidth: 'borderBottomWidth',
  borderColor: 'borderColor',
  borderEndColor: 'borderEndColor',
  borderEndWidth: 'borderEndWidth',
  borderLeftColor: 'borderLeftColor',
  borderLeftWidth: 'borderLeftWidth',
  borderRadius: 'borderRadius',
  borderRightColor: 'borderRightColor',
  borderRightWidth: 'borderRightWidth',
  borderStartColor: 'borderStartColor',
  borderStartWidth: 'borderStartWidth',
  borderStyle: 'borderStyle',
  borderTopColor: 'borderTopColor',
  borderTopEndRadius: 'borderTopEndRadius',
  borderTopLeftRadius: 'borderTopLeftRadius',
  borderTopRightRadius: 'borderTopRightRadius',
  borderTopStartRadius: 'borderTopStartRadius',
  borderTopWidth: 'borderTopWidth',
  borderWidth: 'borderWidth',
  bottom: 'bottom',
  direction: 'direction',
  display: 'display',
  elevation: 'elevation',
  end: 'end',
  flex: 'flex',
  flexBasis: 'flexBasis',
  flexDirection: 'flexDirection',
  flexGrow: 'flexGrow',
  flexShrink: 'flexShrink',
  flexWrap: 'flexWrap',
  height: 'height',
  justifyContent: 'justifyContent',
  left: 'left',
  margin: 'margin',
  marginBottom: 'marginBottom',
  marginEnd: 'marginEnd',
  marginHorizontal: 'marginHorizontal',
  marginLeft: 'marginLeft',
  marginRight: 'marginRight',
  marginStart: 'marginStart',
  marginTop: 'marginTop',
  marginVertical: 'marginVertical',
  maxHeight: 'maxHeight',
  maxWidth: 'maxWidth',
  minHeight: 'minHeight',
  minWidth: 'minWidth',
  opacity: 'opacity',
  overflow: 'overflow',
  padding: 'padding',
  paddingBottom: 'paddingBottom',
  paddingEnd: 'paddingEnd',
  paddingHorizontal: 'paddingHorizontal',
  paddingLeft: 'paddingLeft',
  paddingRight: 'paddingRight',
  paddingStart: 'paddingStart',
  paddingTop: 'paddingTop',
  paddingVertical: 'paddingVertical',
  position: 'position',
  right: 'right',
  rotation: 'rotation',
  scaleX: 'scaleX',
  scaleY: 'scaleY',
  shadowColor: 'shadowColor',
  shadowOffset: 'shadowOffset',
  shadowOpacity: 'shadowOpacity',
  shadowRadius: 'shadowRadius',
  start: 'start',
  top: 'top',
  transform: 'transform',
  transformMatrix: 'transformMatrix',
  translateX: 'translateX',
  translateY: 'translateY',
  width: 'width',
  zIndex: 'zIndex',
};

export const defaultTextStyles: Record<keyof Omit<TextStyle, 'testID'>, string> = {
  alignContent: 'alignContent',
  alignItems: 'alignItems',
  alignSelf: 'alignSelf',
  aspectRatio: 'aspectRatio',
  backfaceVisibility: 'backfaceVisibility',
  backgroundColor: 'backgroundColor',
  borderBottomColor: 'borderBottomColor',
  borderBottomEndRadius: 'borderBottomEndRadius',
  borderBottomLeftRadius: 'borderBottomLeftRadius',
  borderBottomRightRadius: 'borderBottomRightRadius',
  borderBottomStartRadius: 'borderBottomStartRadius',
  borderBottomWidth: 'borderBottomWidth',
  borderColor: 'borderColor',
  borderEndColor: 'borderEndColor',
  borderEndWidth: 'borderEndWidth',
  borderLeftColor: 'borderLeftColor',
  borderLeftWidth: 'borderLeftWidth',
  borderRadius: 'borderRadius',
  borderRightColor: 'borderRightColor',
  borderRightWidth: 'borderRightWidth',
  borderStartColor: 'borderStartColor',
  borderStartWidth: 'borderStartWidth',
  borderStyle: 'borderStyle',
  borderTopColor: 'borderTopColor',
  borderTopEndRadius: 'borderTopEndRadius',
  borderTopLeftRadius: 'borderTopLeftRadius',
  borderTopRightRadius: 'borderTopRightRadius',
  borderTopStartRadius: 'borderTopStartRadius',
  borderTopWidth: 'borderTopWidth',
  borderWidth: 'borderWidth',
  bottom: 'bottom',
  color: 'color',
  direction: 'direction',
  display: 'display',
  elevation: 'elevation',
  end: 'end',
  flex: 'flex',
  flexBasis: 'flexBasis',
  flexDirection: 'flexDirection',
  flexGrow: 'flexGrow',
  flexShrink: 'flexShrink',
  flexWrap: 'flexWrap',
  fontFamily: 'fontFamily',
  fontSize: 'fontSize',
  fontStyle: 'fontStyle',
  fontVariant: 'fontVariant',
  fontWeight: 'fontWeight',
  height: 'height',
  includeFontPadding: 'includeFontPadding',
  justifyContent: 'justifyContent',
  left: 'left',
  letterSpacing: 'letterSpacing',
  lineHeight: 'lineHeight',
  margin: 'margin',
  marginBottom: 'marginBottom',
  marginEnd: 'marginEnd',
  marginHorizontal: 'marginHorizontal',
  marginLeft: 'marginLeft',
  marginRight: 'marginRight',
  marginStart: 'marginStart',
  marginTop: 'marginTop',
  marginVertical: 'marginVertical',
  maxHeight: 'maxHeight',
  maxWidth: 'maxWidth',
  minHeight: 'minHeight',
  minWidth: 'minWidth',
  opacity: 'opacity',
  overflow: 'overflow',
  padding: 'padding',
  paddingBottom: 'paddingBottom',
  paddingEnd: 'paddingEnd',
  paddingHorizontal: 'paddingHorizontal',
  paddingLeft: 'paddingLeft',
  paddingRight: 'paddingRight',
  paddingStart: 'paddingStart',
  paddingTop: 'paddingTop',
  paddingVertical: 'paddingVertical',
  position: 'position',
  right: 'right',
  rotation: 'rotation',
  scaleX: 'scaleX',
  scaleY: 'scaleY',
  shadowColor: 'shadowColor',
  shadowOffset: 'shadowOffset',
  shadowOpacity: 'shadowOpacity',
  shadowRadius: 'shadowRadius',
  start: 'start',
  textAlign: 'textAlign',
  textAlignVertical: 'textAlignVertical',
  textDecorationColor: 'textDecorationColor',
  textDecorationLine: 'textDecorationLine',
  textDecorationStyle: 'textDecorationStyle',
  textShadowColor: 'textShadowColor',
  textShadowOffset: 'textShadowOffset',
  textShadowRadius: 'textShadowRadius',
  textTransform: 'textTransform',
  top: 'top',
  transform: 'transform',
  transformMatrix: 'transformMatrix',
  translateX: 'translateX',
  translateY: 'translateY',
  width: 'width',
  writingDirection: 'writingDirection',
  zIndex: 'zIndex',
};
