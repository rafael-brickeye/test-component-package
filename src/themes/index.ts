export * from './colors';
export * from './sizing';
export * from './spacing';
export * from './utils';
export * from './types';
export * from './constants';
