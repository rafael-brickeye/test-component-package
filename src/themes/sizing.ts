import {FontSizeTypes} from './types';
import {getFontSize} from './utils';

const font: Record<FontSizeTypes, number> = {
  xs: getFontSize(4),
  sm: getFontSize(5),
  md: getFontSize(6),
  lg: getFontSize(7),
  xl: getFontSize(8),
  '2xl': getFontSize(9),
  '3xl': getFontSize(10),
  '4xl': getFontSize(11),
  '5xl': getFontSize(12),
};

export const sizing = {font};
