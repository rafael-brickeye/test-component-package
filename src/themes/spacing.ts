import {SpacingTypes} from './types';
import {generateSpacing} from './utils';

const spacing: Record<SpacingTypes, number> = {
  's-2': generateSpacing(0.5),
  's-4': generateSpacing(1),
  's-8': generateSpacing(2),
  's-12': generateSpacing(3),
  's-16': generateSpacing(4),
  's-20': generateSpacing(5),
  's-24': generateSpacing(6),
  's-28': generateSpacing(7),
  's-32': generateSpacing(8),
  's-36': generateSpacing(9),
  's-40': generateSpacing(10),
};

export {spacing};
