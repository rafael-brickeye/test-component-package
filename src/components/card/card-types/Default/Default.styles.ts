import {StyleSheet} from 'react-native';
import {color, getDefaultStylesFromProps} from '../../../../themes';

const defaultStyles = {
  minHeight: 68,
  borderRadius: 5,
  backgroundColor: color.background.white,
  ...color.shadow.default,
};

const useStyles = ({...props}) => {
  const styles = getDefaultStylesFromProps(props);

  return StyleSheet.create({
    cardContainer: {
      ...defaultStyles,
      ...styles,
    },
  });
};

export default useStyles;
