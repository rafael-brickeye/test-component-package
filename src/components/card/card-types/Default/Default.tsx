import React from 'react';
import {TouchableHighlight, TouchableHighlightProps, ViewStyle} from 'react-native';
import useStyles from './Default.styles';
import {View} from '../../../view';
import {color} from '../../../../themes';

export type DefaultCardPropTypes = {
  onPress: () => void;
  isDisabled: boolean;
} & TouchableHighlightProps &
  ViewStyle;

const DefaultCard: React.FC<DefaultCardPropTypes> = ({isDisabled, onPress, children, style: styleProps, ...props}) => {
  const style = useStyles({isDisabled, ...props});

  return (
    <TouchableHighlight
      disabled={isDisabled}
      underlayColor={color.surface.grey}
      onPress={onPress}
      style={{...style.cardContainer, ...(styleProps as ViewStyle)}}
      {...props}>
      <View opacity={isDisabled ? 0.5 : 1}>{children}</View>
    </TouchableHighlight>
  );
};

export {DefaultCard};
