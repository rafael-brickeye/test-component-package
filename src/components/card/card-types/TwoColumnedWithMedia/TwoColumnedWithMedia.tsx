import React from 'react';
import {Image, ViewStyle, TextStyle} from 'react-native';
import {DefaultCard, DefaultCardPropTypes} from '../Default';
import {View} from '../../../view';
import {Typography} from '../../../typography';
import {color, sizing} from '../../../../themes';
import placeholderIcon from '../../../../../assets/placeholder/card-placeholder.png';

export type TwoColumnedWithMediaCardPropTypes = {
  type: 'two-columned-with-media';
  image?: any;
  middleHeader: string;
  middleHeaderTruncate?: boolean;
  middleHeaderStyle?: TextStyle;
  middleHeaderTestId?: string;
  middleBody: string;
  middleBodyTruncate?: boolean;
  middleBodyStyle?: TextStyle;
  middleBodyTestId?: string;
  rightHeader: string;
  rightHeaderTruncate?: boolean;
  rightHeaderStyle?: TextStyle;
  rightHeaderTestId?: string;
  rightBody: string;
  rightBodyTruncate?: boolean;
  rightBodyStyle?: TextStyle;
  rightBodyTestId?: string;
};

const TwoColumnedWithMediaCard: React.FC<TwoColumnedWithMediaCardPropTypes & DefaultCardPropTypes> = ({
  image,
  middleHeader,
  middleHeaderStyle,
  middleHeaderTruncate,
  middleHeaderTestId,
  middleBody,
  middleBodyStyle,
  middleBodyTruncate,
  middleBodyTestId,
  rightBody,
  rightBodyStyle,
  rightBodyTruncate,
  rightBodyTestId,
  rightHeader,
  rightHeaderStyle,
  rightHeaderTruncate,
  rightHeaderTestId,
  style: styleProps,
  ...props
}) => {
  return (
    <DefaultCard paddingVertical={16} paddingHorizontal={10} style={{...(styleProps as ViewStyle)}} {...props}>
      <View flexDirection={'row'}>
        <Image style={{width: 40, height: 44}} source={image ?? placeholderIcon} />
        <View marginLeft={6} flex={0.65}>
          <Typography
            truncateEnd={middleHeaderTruncate}
            fontWeight={'500'}
            marginBottom={6}
            testID={middleHeaderTestId}
            {...middleHeaderStyle}>
            {middleHeader}
          </Typography>
          <Typography
            truncateEnd={middleBodyTruncate}
            fontSize={sizing.font.sm}
            fontWeight={'500'}
            color={color.text.grey}
            testID={middleBodyTestId}
            {...middleBodyStyle}>
            {middleBody}
          </Typography>
        </View>
        <View flex={0.35} alignItems={'flex-end'}>
          <Typography
            truncateEnd={rightHeaderTruncate}
            fontSize={sizing.font.xs}
            color={color.text.grey}
            marginBottom={6}
            testID={rightHeaderTestId}
            {...rightHeaderStyle}>
            {rightHeader}
          </Typography>
          <Typography
            truncateEnd={rightBodyTruncate}
            fontSize={sizing.font.sm}
            fontWeight={'500'}
            testID={rightBodyTestId}
            {...rightBodyStyle}>
            {rightBody}
          </Typography>
        </View>
      </View>
    </DefaultCard>
  );
};

export {TwoColumnedWithMediaCard};
