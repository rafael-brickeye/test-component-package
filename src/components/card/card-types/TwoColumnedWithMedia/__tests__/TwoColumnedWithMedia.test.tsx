import React from 'react';
import {cleanup, render, screen} from '@testing-library/react-native';
import '@testing-library/jest-native';
import {TwoColumnedWithMediaCard, TwoColumnedWithMediaCardPropTypes} from '../TwoColumnedWithMedia';
import {DefaultCardPropTypes} from '../../Default';
import {color, spacing} from '../../../../../themes';

afterEach(cleanup);

const defaultCardParams = {
  isDisabled: false,
  onPress: () => null,
  middleHeader: 'hello',
  middleHeaderTestId: 'middle-header-text',
  middleBody: 'world',
  middleBodyTestId: 'middle-body-text',
  rightHeader: 'foo',
  rightHeaderTestId: 'right-header-text',
  rightBody: 'bar',
  rightBodyTestId: 'right-body-text',
};
const twoColumnedWithMediaCardTestId = 'two-columned-with-media-card-component';

const TwoColumnedWithMediaCardComponent = (
  props: Partial<TwoColumnedWithMediaCardPropTypes & DefaultCardPropTypes>,
) => <TwoColumnedWithMediaCard type="two-columned-with-media" {...defaultCardParams} {...props} />;

describe('Two Columned With Media Card Component', () => {
  it('should match the existing snapshot', () => {
    const snapshot = render(<TwoColumnedWithMediaCardComponent />).toJSON();

    expect(snapshot).toMatchSnapshot();
  });

  it('should be able to render headers and bodies', () => {
    const {getByTestId} = render(<TwoColumnedWithMediaCardComponent />);

    expect(getByTestId(defaultCardParams.middleHeaderTestId)).toBeVisible();
    expect(getByTestId(defaultCardParams.middleHeaderTestId)).toHaveTextContent(defaultCardParams.middleHeader);
    expect(getByTestId(defaultCardParams.middleBodyTestId)).toBeVisible();
    expect(getByTestId(defaultCardParams.middleBodyTestId)).toHaveTextContent(defaultCardParams.middleBody);
    expect(getByTestId(defaultCardParams.rightHeaderTestId)).toBeVisible();
    expect(getByTestId(defaultCardParams.rightHeaderTestId)).toHaveTextContent(defaultCardParams.rightHeader);
    expect(getByTestId(defaultCardParams.rightBodyTestId)).toBeVisible();
    expect(getByTestId(defaultCardParams.rightBodyTestId)).toHaveTextContent(defaultCardParams.rightBody);
  });

  it('should not be tappable when disabled', () => {
    const {getByTestId} = render(
      <TwoColumnedWithMediaCardComponent testID={twoColumnedWithMediaCardTestId} isDisabled={true} />,
    );

    expect(getByTestId(twoColumnedWithMediaCardTestId)).toBeDisabled();
  });

  it('should be able to override default styling', () => {
    const newStyles = {
      marginVertical: 100,
      paddingHorizontal: 200,
      backgroundColor: 'orange',
    };

    render(<TwoColumnedWithMediaCardComponent testID={twoColumnedWithMediaCardTestId} {...newStyles} />);

    const twoColumnedWithMediaCardComponent = screen.getByTestId(twoColumnedWithMediaCardTestId);

    expect(twoColumnedWithMediaCardComponent).not.toHaveStyle({
      paddingLeft: spacing['s-36'],
      backgroundColor: color.background.white,
    });
    expect(twoColumnedWithMediaCardComponent).toHaveStyle({...newStyles});
  });
});
