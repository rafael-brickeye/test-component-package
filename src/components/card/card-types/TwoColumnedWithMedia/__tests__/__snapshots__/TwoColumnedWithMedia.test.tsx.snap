// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Two Columned With Media Card Component should match the existing snapshot 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#ffffff",
      "borderRadius": 5,
      "elevation": 1,
      "minHeight": 68,
      "paddingHorizontal": 10,
      "paddingVertical": 16,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.18,
      "shadowRadius": 1,
    }
  }
>
  <View
    style={Object {}}
  >
    <View
      flexDirection="row"
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../brickeye-core-ui/assets/placeholder/card-placeholder.png",
          }
        }
        style={
          Object {
            "height": 44,
            "width": 40,
          }
        }
      />
      <View
        flex={0.65}
        marginLeft={6}
        style={
          Object {
            "flex": 0.65,
            "marginLeft": 6,
          }
        }
      >
        <Text
          fontWeight="500"
          marginBottom={6}
          numberOfLines={0}
          style={
            Object {
              "color": "#000000DE",
              "fontSize": 12,
              "fontWeight": "500",
              "marginBottom": 6,
            }
          }
          testID="middle-header-text"
        >
          hello
        </Text>
        <Text
          color="#71727A"
          fontSize={10}
          fontWeight="500"
          numberOfLines={0}
          style={
            Object {
              "color": "#71727A",
              "fontSize": 10,
              "fontWeight": "500",
            }
          }
          testID="middle-body-text"
        >
          world
        </Text>
      </View>
      <View
        alignItems="flex-end"
        flex={0.35}
        style={
          Object {
            "alignItems": "flex-end",
            "flex": 0.35,
          }
        }
      >
        <Text
          color="#71727A"
          fontSize={8}
          marginBottom={6}
          numberOfLines={0}
          style={
            Object {
              "color": "#71727A",
              "fontSize": 8,
              "fontWeight": "normal",
              "marginBottom": 6,
            }
          }
          testID="right-header-text"
        >
          foo
        </Text>
        <Text
          fontSize={10}
          fontWeight="500"
          numberOfLines={0}
          style={
            Object {
              "color": "#000000DE",
              "fontSize": 10,
              "fontWeight": "500",
            }
          }
          testID="right-body-text"
        >
          bar
        </Text>
      </View>
    </View>
  </View>
</View>
`;
